_posts/2017-10-06-few-steps-to-fast-redshift.md

```diff
@@ -0,0 +1,77 @@
+
+**Distribution style and key**
+
+Following Amazon‘s instructions: <cite>[Choose a column with high cardinality in the filtered result set.](http://docs.aws.amazon.com/redshift/latest/dg/c_best-practices-best-dist-key.html)</cite>, it was rather obvious to choose the date of the event as a distribution key as there is only one table and no joins are needed. 
+
+
+
+**Sort key**
+
+One can decide between a compund sort key and an interleaved sort key and choose one or more columns to compose the key. This is best described in [Amazon‘s dosc](http://docs.aws.amazon.com/redshift/latest/dg/t_Sorting_data.html).
+
+<cite>A compound key is made up of all of the columns listed in the sort key definition, in the order they are listed. A compound sort key is most useful when a query's filter applies conditions, such as filters and joins, that use a prefix of the sort keys. The performance benefits of compound sorting decrease when queries depend only on secondary sort columns, without referencing the primary columns. COMPOUND is the default sort type.</cite>
+
+<cite>An interleaved sort gives equal weight to each column, or subset of columns, in the sort key. If multiple queries use different columns for filters, then you can often improve performance for those queries by using an interleaved sort style. When a query uses restrictive predicates on secondary sort columns, interleaved sorting significantly improves query performance as compared to compound sorting.</cite>
+
+Although Amazon‘s documentation is a good source of knowledge, if you really want to understand the idea behind interleaved and compound sort keys, read [this great two-parts article](https://blog.chartio.com/posts/understanding-interleaved-sort-keys-in-amazon-redshift-part-1)
+
+Keeping all this in mind, I decided to try two different approaches:
+ - the date as a sort key – it is the only column to appear in each and every query,
+ - interleaved sort key composed of date and a two columns most often appearing in WHERE clause.
+
+I run a batch of queries (90 sets of 10 queries used by a dashboard with different WHERE clauses) against tables with sort keys defined as above. Here are the results:
+
+![Photo 1](/images/few-steps-to-fast-redshift/chart.png)
+
+The simplest solution turned out to be the best one. It gives about 50% better performance then no sort key and 20% better then the interleaved sort key.
+
+**Other little things**
+
+Another [article](https://aws.amazon.com/blogs/big-data/top-10-performance-tuning-techniques-for-amazon-redshift/) that helped me a lot comes from AWS big data blog and describes ten performance tuning techniques.
+
+I was already familiar with the most of the tips and only two where applying to my case. Nevertheless it is a great article pointing out important improvements accompanied by instructions how to check if you need them and apply them if so.
```

Comment:
"familiar with ~~the~~ most of the tips" ?

Reply:
Corrected.